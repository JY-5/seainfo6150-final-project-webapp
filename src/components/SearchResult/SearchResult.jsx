import React, { useState, useEffect } from 'react'
import styles from "./SearchResult.module.css";

const SearchResult = (props) => {
    const[recipesFound, setRecipesFound] = useState([{title: '', id: '', url: ''}]);
    useEffect(() => {
        const fetchRecipes = async () => {
            // search according to the search keyword
            const response = await fetch("https://run.mocky.io/v3/c4071e7b-ec14-45bc-8ca3-6115283e9689");
            const responseJson = await response.json();
            setRecipesFound(Object.values(responseJson));
            };
            fetchRecipes();
    }, []);

    return (
        <div className={styles.result}>        
            <h3>{props.header}</h3>
            <ul className={styles.list}>
                {recipesFound.map((recipe) => (
                    <li key={recipe.id} className={styles.li}>
                        <img className={styles.image} src={recipe.url} alt={recipe.title} />
                        <p className={styles.title}>{recipe.title}</p>
                    </li>
                ))}
            </ul>
        </div>
    )
}

export default SearchResult