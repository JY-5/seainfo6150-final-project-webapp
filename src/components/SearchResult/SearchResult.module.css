.result {
    margin: 0 2rem;
}

.image {
    width: 10rem;
    height: 10rem;
    border-radius: 5px;
}

.list {
    display: flex;
    flex-flow: row wrap;
    list-style: none;
    width: 100%;
    padding-left: 0;
    margin-top: 0.3rem;
}

.li {
    margin-right: 0.4rem;
    font-family: 'Rubik', sans-serif;
    font-size: 1rem;
    font-weight: 600;
    color: #413e3e;
}

.title {
    margin-top: 0.1rem;
    width: 10rem;
}

.header {
    margin-top: 0.5rem;
    margin-bottom: 0.2rem;
    font-family: 'Rubik', sans-serif;
    /* background-color: #f59542; */
    padding: 0.4rem;
    font-weight: 600;
    font-size: 2rem;
    color: #f59542;
}

.result h3 {
    margin-top: 0.5rem;
    margin-bottom: 0.2rem;
    font-family: 'Rubik', sans-serif;
    padding: 0.4rem;
    font-weight: 600;
    font-size: 2rem;
    color: #539222;
}