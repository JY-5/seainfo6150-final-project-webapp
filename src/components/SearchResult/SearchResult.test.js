import React from "react";
import SearchResult from './SearchResult';
import { render } from "@testing-library/react";

describe("Add SearchResult tests", () => {
    it("renders correctly", () => {
      const { container } = render(<SearchResult/>);
      expect(container).toMatchSnapshot();
    });
});