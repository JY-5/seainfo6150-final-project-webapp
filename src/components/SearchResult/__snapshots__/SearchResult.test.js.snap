// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add SearchResult tests renders correctly 1`] = `
<div>
  <div
    class="result"
  >
    <h3 />
    <ul
      class="list"
    >
      <li
        class="li"
      >
        <img
          alt=""
          class="image"
          src=""
        />
        <p
          class="title"
        />
      </li>
    </ul>
  </div>
</div>
`;
